### ISSUE TITLE
IncompatibleClassChangeError on server start-up

### ISSUE THREAD

**Issue by SlimeDog** [PimvanderLoos/BigDoors#14]
Paper 1.20.1-126
BigDoors 0.1.8.47 (b1141)

This did not occur previously, so I suspect Paper has changed a class definition.

On server start-up, the following
```
[17:51:57] [Server thread/WARN]: [BigDoors] Task #24457 for BigDoors vAlpha 0.1.8.47 (b1141) generated an exception
java.lang.IncompatibleClassChangeError: class nl.pim16aap2.bigDoors.NMS.CustomCraftFallingBlock_V1_20_R1 overrides final method org.bukkit.craftbukkit.v1_20_R1.entity.CraftEntity.getType()Lorg/bukkit/entity/EntityType;
        at java.lang.ClassLoader.defineClass1(Native Method) ~[?:?]
        at java.lang.ClassLoader.defineClass(ClassLoader.java:1013) ~[?:?]
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150) ~[?:?]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:229) ~[paper-api-1.20.1-R0.1-SNAPSHOT.jar:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:588) ~[?:?]
        at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:155) ~[paper-api-1.20.1-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:150) ~[paper-api-1.20.1-R0.1-SNAPSHOT.jar:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[?:?]
        at nl.pim16aap2.bigDoors.NMS.FallingBlockFactory_V1_20_R1.fallingBlockFactory(FallingBlockFactory_V1_20_R1.java:21) ~[BigDoors-0.1.8.47-b1141.jar:?]
        at nl.pim16aap2.bigDoors.moveBlocks.VerticalMover.createAnimatedBlocks(VerticalMover.java:114) ~[BigDoors-0.1.8.47-b1141.jar:?]
        at org.bukkit.craftbukkit.v1_20_R1.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.20.1.jar:git-Paper-126]
        at org.bukkit.craftbukkit.v1_20_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:480) ~[paper-1.20.1.jar:git-Paper-126]
        at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1480) ~[paper-1.20.1.jar:git-Paper-126]
        at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:447) ~[paper-1.20.1.jar:git-Paper-126]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1394) ~[paper-1.20.1.jar:git-Paper-126]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1171) ~[paper-1.20.1.jar:git-Paper-126]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:318) ~[paper-1.20.1.jar:git-Paper-126]
        at java.lang.Thread.run(Thread.java:1623) ~[?:?]
[17:52:08] [Server thread/INFO]: [BigDoors] Door 6 "KingsLanding-S-portal-portcullis" cannot be opened as it is already open!
[17:52:08] [Server thread/WARN]: [BigDoors] Task #24513 for BigDoors vAlpha 0.1.8.47 (b1141) generated an exception
java.lang.IncompatibleClassChangeError: class nl.pim16aap2.bigDoors.NMS.CustomCraftFallingBlock_V1_20_R1 overrides final method org.bukkit.craftbukkit.v1_20_R1.entity.CraftEntity.getType()Lorg/bukkit/entity/EntityType;
        at nl.pim16aap2.bigDoors.NMS.FallingBlockFactory_V1_20_R1.fallingBlockFactory(FallingBlockFactory_V1_20_R1.java:21) ~[BigDoors-0.1.8.47-b1141.jar:?]
        at nl.pim16aap2.bigDoors.moveBlocks.CylindricalMover.createAnimatedBlocks(CylindricalMover.java:167) ~[BigDoors-0.1.8.47-b1141.jar:?]
        at org.bukkit.craftbukkit.v1_20_R1.scheduler.CraftTask.run(CraftTask.java:101) ~[paper-1.20.1.jar:git-Paper-126]
        at org.bukkit.craftbukkit.v1_20_R1.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:480) ~[paper-1.20.1.jar:git-Paper-126]
        at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1480) ~[paper-1.20.1.jar:git-Paper-126]
        at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:447) ~[paper-1.20.1.jar:git-Paper-126]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1394) ~[paper-1.20.1.jar:git-Paper-126]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1171) ~[paper-1.20.1.jar:git-Paper-126]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:318) ~[paper-1.20.1.jar:git-Paper-126]
        at java.lang.Thread.run(Thread.java:1623) ~[?:?]
```

**Comment by Hellcode48** [comment]
Having same problem on same version of Paper Spigot

**Comment by Honsal** [comment]
same problem on purpur

**Comment by PimvanderLoos** [comment]
A fix for this issue is now available [here](https://pim16aap2.nl/BigDoors/b1142/).

Note that, as stated in the changelog, this build only works on recent builds of Spigot/Paper/etc. Basically, it only works on versions where it threw the exception above before this patch.

**Comment by SlimeDog** [comment]
Confirmed fixed with BigDoors 0.1.8.47-b1142. Thanks.
